src/Features/CSharp/Portable/MoveToResx/CSharpMoveToResxCodeFixProvider.cs

```diff
@@ -0,0 +1,235 @@
+                .FirstOrDefault(e => (string)e.Attribute("name") == name);
+
+            if (dataElement == null)
+            {
+                dataElement = new XElement("data",
+                    new XAttribute("name", name),
+                    new XAttribute(XNamespace.Xml + "space", "preserve"),
+                    new XElement("value", value));
+                xdoc.Root.Add(dataElement);
+            }
+            else
+            {
+                dataElement.Element("value")!.Value = value;
+            }
+
+            // 4. Serialize the updated XDocument
+            string updatedResxText;
+            using (var sw = new StringWriter())
+            {
+                xdoc.Save(sw);
+                updatedResxText = sw.ToString();
+            }
+
+            // 5. Create new SourceText
+            var newResxSourceText = SourceText.From(updatedResxText, resxSourceText.Encoding);
+
+            // 6. Update the AdditionalDocument in the Solution (in memory)
+            var newSolution = document.Project.Solution.WithAdditionalDocumentText(resxFile.Id, newResxSourceText);
+
+            // 7. Replace the string literal in the code with a resource reference, including namespace if available
+            var resourceClass = Path.GetFileNameWithoutExtension(resxFile.Name);
+            var ns = document.Project.DefaultNamespace;
+            string resourceAccessString = !string.IsNullOrEmpty(ns)
+                ? $"{ns}.{resourceClass}.{name}"
+                : $"{resourceClass}.{name}";
+
+            // After replacing the string literal:
+            var resourceAccess = SyntaxFactory.ParseExpression(resourceAccessString).WithTriviaFrom(stringLiteral);
+
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var newRoot = root.ReplaceNode(stringLiteral, resourceAccess);
+            newSolution = newSolution.WithDocumentSyntaxRoot(document.Id, newRoot);
+
+            // 8. Return the updated Solution
+            return newSolution;
+        }
+
+        private static string ToDeterministicResourceKey(string value, int maxWords = 6)
+        {
+            if (string.IsNullOrWhiteSpace(value))
+                return "emptyString";
+
+            // Split the string into words using non-alphanumeric as delimiters
+            var words = new List<string>();
+            var sb = new StringBuilder();
+            foreach (char c in value)
+            {
+                if (char.IsLetterOrDigit(c))
+                {
+                    sb.Append(c);
+                }
+                else if (sb.Length > 0)
+                {
+                    words.Add(sb.ToString().ToLowerInvariant());
+                    sb.Clear();
+                }
+            }
+            if (sb.Length > 0)
+                words.Add(sb.ToString().ToLowerInvariant());
+
+            // Limit to maxWords
+            if (words.Count > maxWords)
```

Comment:
this seems quite low.  What i'd probably do instead is make the resource string as long as needed without going over a max length (like 60 chars) vs a max number of words (which could be any any length).